src/cli-validator.js

```diff
@@ -12,18 +12,30 @@ function validateArgv(argv) {
   }
 }
 
-function validateIntOption(value, option) {
-  if (!/^\d+$/.test(value)) {
+function getOptionName(type, option) {
+  return type === "cli" ? `--${option.name}` : kebabToCamel(option.name);
+}
+
+function kebabToCamel(str) {
+  return str.replace(/-([a-z])/g, (_, char) => char.toUpperCase());
+}
+
+function validateIntOption(type, value, option) {
+  if (!/^\d+$/.test(value) || (type === "api" && typeof value !== "number")) {
```

Comment:
I like how you've really thought things through and found even more edge cases we were missing!